src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesRangeResult.cs

```diff
@@ -10,4 +10,9 @@ public class TimeSeriesRangeResult
         public long? TotalResults;
         internal string Hash;
     }
+
+    public class TimeSeriesRangeResult<TValues> : TimeSeriesRangeResult where TValues : TimeSeriesEntry
```

Comment:
Would it help us to have it the other way around?

The `TimeSeriesRangeResult` would inherit from `TimeSeriesRangeResult<TimeSeriesEntry>`? Would that help us to avoid 'new' ?

Reply:
We can't.
Because we can't cast from base class to the derived.